distro/run/assembly/resources/start.bat

```diff
@@ -43,26 +47,47 @@ IF [%~1]==[--webapps] (
 
 IF [%~1]==[--rest] (
   SET optionalComponentChosen=true
+  SET restChosen=true
   SET classPath=%restPath%,%classPath%
   ECHO REST API enabled
 )
 
+IF [%~1]==[--swaggerui] (
+  SET optionalComponentChosen=true
+  SET swaggeruiChosen=true
+  SET classPath=%swaggerPath%,%classPath%
+  ECHO Swagger UI enabled
+)
+
 IF [%~1]==[--production] (
+  SET productionChosen=true
   SET configuration=%BASEDIR%configuration\production.yml
 )
 
-
 SHIFT
 GOTO Loop
 :Continue
 
-REM if neither REST nor Webapps are explicitly chosen, enable both
+REM if neither REST, Webapps nor are explicitly chosen, enable all three
 IF [%optionalComponentChosen%]==[false] (
+  SET restChosen=true
   ECHO REST API enabled
   ECHO WebApps enabled
+  IF [%productionChosen%]==[false] (
+    SET swaggeruiChosen=true
+    ECHO Swagger UI enabled
+    SET classPath=%swaggerPath%,%classPath%
+  )
   SET classPath=%webappsPath%,%restPath%,%classPath%
 )
```

Comment:
Setting the `classPath` in the nested IF will not be considered two lines down unless you explicitly use delayed expansion and resolve the variable with `!classPath!`. If you're interested, you can have a look here for some explanation: https://superuser.com/a/78509.

```suggestion
setlocal enabledelayedexpansion
IF [%optionalComponentChosen%]==[false] (
  SET restChosen=true
  ECHO REST API enabled
  ECHO WebApps enabled
  IF [%productionChosen%]==[false] (
    SET swaggeruiChosen=true
    ECHO Swagger UI enabled
    SET classPath=%swaggerPath%,%classPath%
  )
  SET classPath=%webappsPath%,%restPath%,!classPath!
)
setlocal disabledelayedexpansion
```